src/webgpu/api/validation/queue/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,409 @@
+    dimension: GPUTextureDimension,
+    format: GPUTextureFormat,
+    usage: GPUTextureUsageFlags,
+    mipLevel: number,
+    origin: GPUOrigin3D,
+    destroyed: boolean
+  ): GPUTextureCopyView {
+    let size: GPUExtent3D = { width: 0, height: 0, depth: 0 };
+    switch (dimension) {
+      case '1d':
+        size = { width: DEFAULT_WIDTH, height: 1, depth: 1 };
+        break;
+      case '2d':
+        size = { width: DEFAULT_WIDTH, height: DEFAULT_HEIGHT, depth: 1 };
+        break;
+      case '3d':
+        size = { width: DEFAULT_WIDTH, height: DEFAULT_HEIGHT, depth: DEFAULT_DEPTH };
+        break;
+      default:
+        unreachable('unknown texture dimension');
+    }
+
+    return this.getTextureCopyView(
+      size,
+      sampleCount,
+      dimension,
+      format,
+      usage,
+      mipLevel,
+      origin,
+      destroyed
+    );
+  }
+
+  // Generate dst texture copy view with default width, height and correct configs.
+  getDefaultTextureCopyView(dimension: GPUTextureDimension): GPUTextureCopyView {
+    return this.getTextureCopyViewWithDefaultSize(
+      1,
+      dimension,
+      'bgra8unorm',
+      GPUTextureUsage.COPY_DST,
+      0,
+      { x: 0, y: 0, z: 0 },
+      false
+    );
+  }
+
+  testRun(
+    imageBitmapCopyView: GPUImageBitmapCopyView,
+    textureCopyView: GPUTextureCopyView,
+    copySize: GPUExtent3D,
+    success: boolean,
+    exceptionName?: string
+  ): void {
+    // CopyImageBitmapToTexture will generate two types of errors. One is exceptions generated out
+    // of Dawn and the other is validation errors from Dawn.
+    if (exceptionName && exceptionName.length !== 0) {
+      this.shouldThrow(exceptionName, () => {
+        this.device.defaultQueue.copyImageBitmapToTexture(
+          imageBitmapCopyView,
+          textureCopyView,
+          copySize
+        );
+      });
+    } else {
+      this.expectValidationError(() => {
+        this.device.defaultQueue.copyImageBitmapToTexture(
+          imageBitmapCopyView,
+          textureCopyView,
+          copySize
+        );
+      }, !success);
+    }
+  }
+}
+
+export const g = makeTestGroup(CopyImageBitmapToTextureTest);
+
+g.test('copyImageBitmapToTexture_source_imageBitmap_copy_view')
+  .params(poptions('closed', [true, false]))
+  .fn(async t => {
+    const { closed } = t.params;
+    const src = await t.getImageBitmapCopyView(
+      DEFAULT_WIDTH,
+      DEFAULT_HEIGHT,
+      { x: 0, y: 0 },
+      closed
+    );
+    const dst = t.getDefaultTextureCopyView('2d');
+    const copySize = { width: 1, height: 1, depth: 1 };
+    const exceptionName = closed ? 'InvalidStateError' : '';
+
+    t.testRun(src, dst, copySize, !closed, exceptionName);
+  });
+
+g.test('copyImageBitmapToTexture_destination_texture_copy_view')
+  .params(
+    params()
+      .combine(poptions('destroyed', [true, false]))
+      .combine(poptions('dimension', ['2d'] as const))
+      .combine(poptions('usage', kTextureUsages))
```

Comment:
nit: some of these tests could actually be split. Right now it does a lot of combination for almost 0 increased coaverage.

Reply:
Agree, most of these rules are very separate. For example, implementations wouldn't be likely to catch a destroyed texture, and catch a multisampled texture, but not catch a destroyed multisampled texture.

Reply:
Yes, and sry for that. Your points is correct because we don't know the wrong param is caught by corresponding validation rule in such a big combination. Working on address this.

Reply:
I don't think that's an issue, as I think that even with the big combination, there will be one test that tests ONLY each validation rule. But it would reduce the runtime and reduce the big list of validation computations in the test function (which would be easier to understand in isolation, one per test).

Reply:
Ack. And reshape the tests.